packages/rancher-pushprox/charts/templates/_helpers.tpl

```diff
@@ -84,4 +84,21 @@ k8s-app: {{ template "pushProxy.proxy.name" . }}
 app: {{ template "pushprox.serviceMonitor.name" . }}
 release: {{ .Release.Name | quote }}
 {{ template "pushProxy.commonLabels" . }}
+{{- end -}}
+
+{{- define "pushProxy.serviceMonitor.endpoints" -}}
+{{- $proxyURL := (include "pushProxy.proxyUrl" .) -}}
+{{- $useHTTPS := .Values.clients.https.enabled -}}
+{{- $endpoints := .Values.serviceMonitor.endpoints }}
+{{- range $endpoints }}
+{{- $_ := set . "proxyUrl" $proxyURL }}
+{{- if $useHTTPS -}}
+{{- if (hasKey . "params") }}
+{{- $_ := set (get . "params") "_scheme" (list "https") }}
```

Comment:
What does this line do exactly? Cross-referencing with sprig docs and want to make sure I understand it correctly.

Reply:
Overrides proxyUrl and params._scheme to the values we expect for the clients. Otherwise, respects the values provided.